### PR TITLE
VxDesign: Add persistent contest nav list to Contests screen

### PR DESCRIPTION
**Pull request by kofi-q**
## Overview

_Bit of indentation thrash in `contests_screen` - might help to ignore whitespace_

https://github.com/votingworks/vxsuite/issues/7266

Combine the contest list with the view/edit pane, in preparation for adding [in-context audio editing](https://www.figma.com/design/g5S6rv7kYH8kfEomZTzZqr/-VxDesign--In-Context-Audio-Editing?node-id=1-19&t=AjIijIn8mKGRqctm-0).

Now keeping the contest list and related action/menu bar persistent while navigation through/editing contests.
Preserves reordering functionality - though, would like to eventually follow up with an update to move the reorder button visually closer to the list, now that it's a sidebar.

### Mainline Functionality

https://github.com/user-attachments/assets/2407a1cb-be6e-42b6-9f49-46a6f78e402c

https://github.com/user-attachments/assets/1e4ef9e0-eda5-4830-b0ca-59c9d60d72fb

### Layout Edge Cases

https://github.com/user-attachments/assets/a30a51cc-15ea-4593-8fa9-e07741a0a88c

## Testing Plan
- Updated + additional unit tests
- Cross-browser checks in Chrome, FF, Safari

## Checklist

- [x] I have prefixed my PR title with "VxDesign: ", "VxPollBook: ", or "HWTA: " if my change is specific to one of those products.
- [x] I have added the "user_facing_change" label to this PR, if relevant, to automate an announcement in #machine-product-updates.

### CODE REVIEW

**Comment by kofi-q** [comment]
> The scroll bar for the contest list feels a bit off to me. I think it's because there's a border on its left but not on its right, so it visually seems to be part of the main panel rather than the list.

Yeah, makes sense - we could try moving it to within the list box itself and see. Cross-browser scrollbar styling is iffy AFAICT and I worry about the browser putting a visible track bar along the list items, but worth a shot.

> Are we doing purple scroll bars across the board, or just here? I generally prefer low key gray scroll bars, but I was thinking maybe you made them purple to draw attention to the scrollability. Is that the thinking?

Currently have it applied along the fixed forms and the sidebar lists (some examples in the Figma mocks) - and yeah, the initial motivation was to provide some indication scroll was available and then ended up just liking the way they looked. Will switch to gray.

> It feels a little odd that the bottom actions bar in the main panel has a top border that doesn't connect to the edges of the panel, given that the rest of the boundaries in the page go all the way.
I wonder if we moved the reorder button to the bottom of the list then we could have a shared footer across both (with a vertical divider between the two sections). Though that might be confusing in new ways.

Yeah, we can iterate some more on that. Probably have a few options at our disposal - maybe even move the form into more of a bordered panel (card-like) for clearer separation. Like you mentioned, moving to a reorder actions footer might open up some opportunities, but gets tricky, given the two panes scroll separately.

Heads-up, you mentioned none blocking, but still pending approval - LMK if you'd prefer any of these addressed in the PR vs later. (Besides the the scroll color change, which is quick enough to do now)

**Review by jonahkagan — COMMENTED** [review]
Nice work!

**Review by jonahkagan — APPROVED** [review]
Accidentally did Comment instead of Approve on my review, so now approving

**Comment by jonahkagan** [comment]
> Heads-up, you mentioned none blocking, but still pending approval - LMK if you'd prefer any of these addressed in the PR vs later. (Besides the the scroll color change, which is quick enough to do now)

Happy to have design polish folded in now or later at your discretion. I don't think any of my feedback needs to be addressed for the screen to be useable.

File: apps/design/frontend/src/contests_screen.tsx [hunk 1]
```diff
@@ -39,52 +32,79 @@ export function ContestsScreen(): JSX.Element {
...
+const FILTER_ALL = 'all';
+const FILTER_NONPARTISAN = 'nonpartisan';
+
+const Viewport = styled(FixedViewport)`
+  display: grid;
+  grid-template-rows: min-content 1fr;
+`;
+
+const ListActionsRow = styled.div`
+  border-bottom: 1px solid #aaa;
```
[the remark this answers is not in the document]
Comment (kofi-q) [reply apps/design/frontend/src/contests_screen.tsx]: Ah, thanks - leftover from prototyping

File: apps/design/frontend/src/contest_list.tsx [hunk 2]
```diff
@@ -0,0 +1,299 @@
...
+import { cssThemedScrollbars } from './scrollbars';
+import { ElectionIdParams, routes } from './routes';
+
+const CLASS_REORDER_BUTTON = 'contestReorderButton';
+const CLASS_SUBLIST_ITEMS = 'contestSublistItems';
+
+const Item = styled.li`
+  align-items: center;
+  border-bottom: var(--contest-list-border);
+  border-bottom-style: dashed;
```
[the remark this answers is not in the document]
Comment (kofi-q) [reply apps/design/frontend/src/contest_list.tsx]: Reasoning: been using dashed borders as a way to mute less important lines without overly reducing their contrast, which is what initially led to using them for the disabled state in the machine UIs. In this case, the repeated solid lines looked a little busy/distracting for me as I was iterating and made it tougher to focus on the text within, but the delineation still felt vaguely useful, so the dashed lines seemed a good compromise.
That being said, not strongly tied to it - will switch
